### PR TITLE
batch inference in evaluator.py available(single gpu)

### PR DESCRIPTION
**Pull request by yaoguany**
mainly changed the evaluator.py,some minor changes in hf_deocder_model.py

### CODE REVIEW

**Comment by SHUMKASHUN** [comment]
Thanks for @yaoguany 's contribution on batch inference which significantly boost the inference speed !
Here are some questions I have :

### `src/lmflow/models/hf_decoder_model.py`
- [Question] line 337-339 : I am wondering if this works for all cases. If yes, we may move it to the `__init__`, if not, we may let padding_side/pad_token be an argument.
- [Remind] line 340: Notice that the return value of function `encode` is `tokeniser()` (a BatchEncoding class) instead of `tokenise.encode()` (list of token ids). 
- [Remind] line 364: Same as above, return value is batch decode instead of decode.
For above two remind, have you checked that anywhere else in the repo which use this two apis is still woking properly? If not, please remind anyone related to that part to change their codes.

### `General`
- [Question] Since currently batch inference only support single GPU, shall we merge this code while also preserve the original inference code which support multi-gpus ? @research4pan @shizhediao 
- [Question] Have this new batch inference pass the regression/unit test and obtain similar accuracy?

The remaining LGTM! Thanks again for this contribution.

**Comment by yaoguany** [comment]
### src/lmflow/models/hf_decoder_model.py
- [Question] line 337-339 :Referring to the link below,I think for all decoder-only models, the answer for pad_site is yes. As for pad_token, some decoder models like llama-7b-hf don't have it's pad_token ,so we need to specify it before encode the batch.
https://github.com/huggingface/transformers/issues/14521#issuecomment-990895170
- [Remind] line 364: I checked evaluator.py and inferencer.py in pipeline have the correct output. Inferencer.py's output will just be  the same as before if batchsize=1.
### General
- [Question] Have this new batch inference pass the regression/unit test and obtain similar accuracy?
According to the link below, there will be a little differences of the inference results between padding input and original input, because the mask_attention after softmax is close to but not ZERO.I repeated some experiments on MedQA-USMLE of llama7b-lora-medical, the final accuracy deviation fluctuates around 0.5%.I think it's an acceptable behavior.
https://github.com/huggingface/transformers/issues/14521#issuecomment-992428858

**Review by research4pan — CHANGES_REQUESTED** [review]
Thanks for this meaningful contribution! 👍 Batch encode/decode will automatically handle the padding problem, and will be very important for speeding up the evaluation process.

The major part of current implementation looks good to me. Thanks! The pull request can be merged after 1) fixing the following issues, 2) passing all unittests and 3) successfully running all scripts.

### `examples/evaluate.py`

- [Style] Better remove commented-out codes so the interface can be clean for users.

### `scripts/run_evaluation_with_lorash`

- [Compatibility] Please keep this file unchanged 🙏 , since 1) The script behavior will change and it is unfriendly to our users; 2) 7b model may be a bit large as an example; 3) the lora checkpoint is not available for most users.

### `src/lmflow/args.py`

- [Compatibility] line 451: Please set default value of `batch_size` to 1, so it is backward-compatible with our old code. Thanks 🙏 

### `src/lmflow/models/hf_decoder_model.py`

- [Bug] line 337-339: Setting of the `tokenizer.pad_token_id` should go to `__init__`. Otherwise the tokenizer behavior will be different before and after the `.encode`, which may implicitly introduce subtle bugs hard to locate.

### `src/lmflow/pipeline/evaluator.py`

- [Style] line 159: Better if `decode_input` -> `decoded_input`

### `src/lmflow/pipeline/inferencer.py`

- [Bug] line 128: `attenion_mask` -> `attention_mask`
